driver-async/src/examples/gridfs/GridFSTour.java

```diff
@@ -0,0 +1,303 @@
+import com.mongodb.client.gridfs.model.GridFSFile;
+import com.mongodb.client.gridfs.model.GridFSUploadOptions;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.mongodb.async.client.gridfs.helpers.AsyncStreamHelper.toAsyncInputStream;
+import static com.mongodb.async.client.gridfs.helpers.AsynchronousChannelHelper.channelToOutputStream;
+import static com.mongodb.client.model.Filters.eq;
+
+/**
+ * The GridFS code example see: https://mongodb.github.io/mongo-java-driver/3.1/driver/reference/gridfs
+ */
+public final class GridFSTour {
+
+    /**
+     * Run this main method to see the output of this quick example.
+     *
+     * @param args takes an optional single argument for the connection string
+     * @throws FileNotFoundException if the sample file cannot be found
+     * @throws IOException if there was an exception closing an input stream
+     * @throws InterruptedException if a latch was interrupted
+     */
+    public static void main(final String[] args) throws FileNotFoundException, InterruptedException, IOException {
+        MongoClient mongoClient;
+
+        if (args.length == 0) {
+            // connect to the local database server
+            mongoClient = MongoClients.create();
+        } else {
+            mongoClient = MongoClients.create(args[0]);
+        }
+
+        // get handle to "mydb" database
+        MongoDatabase database = mongoClient.getDatabase("mydb");
+        final CountDownLatch dropLatch = new CountDownLatch(1);
+        database.drop(new SingleResultCallback<Void>() {
+            @Override
+            public void onResult(final Void result, final Throwable t) {
+                dropLatch.countDown();
+            }
+        });
+        dropLatch.await();
+
+        GridFSBucket gridFSBucket = GridFSBuckets.create(database);
+
+        /*
+         * UploadFromStream Example
+         */
+        // Get the input stream
+        final AsyncInputStream streamToUploadFrom = toAsyncInputStream("Hello World".getBytes(StandardCharsets.UTF_8));
+
+        // Create some custom options
+        GridFSUploadOptions options = new GridFSUploadOptions()
+                .chunkSizeBytes(1024)
+                .metadata(new Document("type", "presentation"));
+
+        final AtomicReference<ObjectId> fileIdRef = new AtomicReference<ObjectId>();
+        final CountDownLatch uploadLatch = new CountDownLatch(1);
+        gridFSBucket.uploadFromStream("mongodb-tutorial", streamToUploadFrom, options, new SingleResultCallback<ObjectId>() {
```

Comment:
Updated the content to be "MongoDB Tutorial.."  This is so its follows the walkthrough documentation in `gridfs/index.md` as closely as it can (without an actual pdf).